pkg/endpoint/bpf.go

```diff
@@ -735,8 +735,18 @@ func (e *Endpoint) runPreCompilationSteps(regenContext *regenerationContext, rul
 	stats := &regenContext.Stats
 	datapathRegenCtxt := regenContext.datapathRegenerationContext
 
+	// regenerate policy without holding the lock.
+	// This is because policy generation needs the ipcache to make progress, and the ipcache needs to call
+	// endpoint.ApplyPolicyMapChanges()
```

Comment:
Care should be taken with applying incremental policy updates while calculating the endpoint policy, as they are functionally interdependent via code like this:
```
			if e.desiredPolicy == e.realizedPolicy {
				// Any map updates when a new policy has not been calculated are taken care by incremental map updates.
				continue
			}
```

In the worst case an incremental change may get "lost" due to (full) policy update not seeing it yet, but then the incremental change not being pushed to the datapath due to parallel executing full policy update apparently having done a "full" update.

Reply:
I checked the code again, did not recall that we pull in incremental changes that may have happened during the policy computation right before we sync the bpf policy map.